src/app-host/app-host.js

```diff
@@ -5,93 +5,187 @@ var Messages = require('messages');
+                plugins[pluginId] = plugin;
+            }
+            if (clobberScope) {
+                clobber(plugin, clobberScope, clobberToPluginMap, pluginId);
+            }
+        }
+    });
+}
+
 function setCordova(originalCordova) {
+    var channel,
+        platform,
+        platformBoostrap;
+
     if (cordova) {
         return;
     }
 
     cordova = originalCordova;
 
-    oldExec = cordova.require('cordova/exec');
     cordova.define.remove('cordova/exec');
     cordova.define('cordova/exec', function (require, exports, module) {
         module.exports = exec;
     });
 
-    // android platform has its own specific initialization
-    // so to emulate it, we need to fake init function
-    if (cordova.platformId === 'android') {
-        exec.init = function () {
-            cordova.require('cordova/channel').onNativeReady.fire();
-        };
-    }
-
-    // windows phone platform fires 'deviceready' event from native component
-    // so to emulate it we fire it in the bootstrap function (similar to ios)
-    if (cordova.platformId === 'windowsphone') {
-        cordova.require('cordova/platform').bootstrap = function () {
-            cordova.require('cordova/channel').onNativeReady.fire();
-        };
-    }
+    platform = cordova.require('cordova/platform');
+    platformBootstrap = platform.bootstrap;
+    platform.bootstrap = function () {
+    };
 
     // default Windows bootstrap function tries to load WinJS which is not
     // available and not required in simulation mode so we override bootstrap
     if (cordova.platformId === 'windows') {
-        cordova.require('cordova/platform').bootstrap = function () {
+        platformBootstrap = function () {
             cordova.require('cordova/modulemapper')
                 .clobbers('cordova/exec/proxy', 'cordova.commandProxy');
 
-            cordova.require('cordova/channel').onNativeReady.fire();
         };
     }
-}
 
-function getCordova() {
-    return cordova;
-}
+    channel = cordova.require('cordova/channel');
+
+    // define our own channel to delay the initialization until sim-host tells
+    // us everything's ready (fired in 'start' event handler).
+    channel.createSticky('onCordovaSimulateReady');
+    channel.waitForInitialization('onCordovaSimulateReady');
+
+    // firing of onNativeReady is delayed until SIM_HOST tells us it's ready
+    socket.once('init', function () {
+        // sim-host is ready, register exec handlers, fire onNativeReady and send
+        // the list of plugins
+        socket.on('exec-success', function (data) {
+            console.log('exec-success:');
+            console.log(data);
+            var execCacheInfo = execCache[data.index];
+            if (execCacheInfo.success) {
+                execCacheInfo.success(data.result);
+            }
+        });
+
+        socket.on('exec-failure', function (data) {
+            console.log('exec-failure:');
+            console.log(data);
```

Comment:
I don't think cordova keeps track of that (among the implementation for the platforms I've seen). We could though, but it will not be part of this PR. We should have a "verbose" mode for `APP_HOST` where things such as exec traces and unreturned calls are printed.